### PR TITLE
Saul 42

### PR DESCRIPTION
**Pull request by Renkon**
Create done 

+ extra

error handling for all

### CODE REVIEW

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_help.py [hunk 1]
```diff
@@ -177,6 +177,17 @@
...
 """
 
+helps['ams content-key-policy'] = """
+    type: group
+    short-summary: Manage content key policies for an Azure Media Services account.
+"""
+
+helps['ams content-key-policy create'] = """
+    type: command
+    short-summary: Create a new content key policy resource.
```
Comment (damatojuan) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_help.py]: I would remove the last word 'resource' as others create commands don't have it

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_params.py [hunk 2]
```diff
@@ -131,6 +131,19 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
...
+        c.argument('content_key_policy_name', name_arg_type)
+        c.argument('description', help='The content key policy description.')
+        c.argument('clear_key_configuration',
+                   action='store_true',
+                   arg_group='Simple Policy Options',
+                   help='Use clear key configuration.')
+        c.argument('open_restriction',
+                   action='store_true',
+                   arg_group='Simple Policy Options',
+                   help='Use open restriction.')
```
Comment (damatojuan) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_params.py]: I would add something like `License or key will be delivered on every request.` to give CLI user more details about the flag.

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_params.py [hunk 3]
```diff
@@ -131,6 +131,19 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
...
         c.argument('delete', action='store_true', help='Delete the job being cancelled.')
 
+    with self.argument_context('ams content-key-policy') as c:
+        c.argument('account_name', account_name_arg_type)
+        c.argument('content_key_policy_name', name_arg_type)
+        c.argument('description', help='The content key policy description.')
+        c.argument('clear_key_configuration',
+                   action='store_true',
+                   arg_group='Simple Policy Options',
+                   help='Use clear key configuration.')
```
Comment (damatojuan) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_params.py]: `Use Clear Key configuration, a.k.a AES encryption. It's intended for non-DRM keys.` (taken from SDK doc & ms web docs)

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_params.py [hunk 4]
```diff
@@ -131,6 +131,19 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
     with self.argument_context('ams job cancel') as c:
         c.argument('delete', action='store_true', help='Delete the job being cancelled.')
 
+    with self.argument_context('ams content-key-policy') as c:
+        c.argument('account_name', account_name_arg_type)
+        c.argument('content_key_policy_name', name_arg_type)
+        c.argument('description', help='The content key policy description.')
+        c.argument('clear_key_configuration',
+                   action='store_true',
+                   arg_group='Simple Policy Options',
```
Comment (damatojuan) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_params.py]: I prefer something like `Basic Policy Options` ... anyway, we will have to double check for IR

File: src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/content_key_policy.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    default_configuration = None
+    default_restriction = None
+
+    if clear_key_configuration:
+        default_configuration = ContentKeyPolicyClearKeyConfiguration()
+
+    if open_restriction:
+        default_restriction = ContentKeyPolicyOpenRestriction()
+
+    options = [ContentKeyPolicyOption(name='Basic policy option',
```
Comment (damatojuan) [inline src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/operations/content_key_policy.py]: why don't parametrize the option name?

**Review by damatojuan — CHANGES_REQUESTED** [review]

**Review by damatojuan — APPROVED** [review]
lgtm

**Review by NahuelBerg — APPROVED** [review]
Good job Ferchu

**Pull request merged**